ethereum-rpc/src/constructors/express/routes/api/v1/tokens.js

```diff
@@ -4,1236 +4,1293 @@ const { body, query, param, validationResult } = require("express-validator");
+          return res.status(400).json({ errors: errors.array() });
+        }
+        const { address = "*" } = req.params;
+        let { tokenAddress = "" } = { ...req.query };
+        const tokenContractAddress = prefixContractAddress(
+          tokenAddress.toString("hex")
+        );
+        if (!web3.utils.isAddress(tokenContractAddress)) {
+          return res.status(500).json({ error: "invalid contract address" });
+        }
+        const contractAbi = abi.APIS_ERC721;
+        const contract = await contractInstance(
+          contractAbi,
+          tokenContractAddress,
+          address
+        );
+        contract.methods
+          .name()
+          .call()
+          .then((result) => {
+            res.status(200).json({ result });
+          });
+      } catch (e) {
+        console.error(e);
+        return res.status(500).json({ errors: [e] });
+      }
     }
-    const contractAbi = abi.APIS_ERC721;
-    const contract = await contractInstance(
-     contractAbi,
-     tokenContractAddress,
-     address
-    );
-    contract.methods
-     .totalSupply()
-     .call()
-     .then((result) => {
-      res.status(200).json({ result });
-     });
-   } catch (e) {
-    console.error(e);
-    return res.status(500).json({ errors: [e] });
-   }
-  }
- );
+  );
 
- router.get(
-  "/:address/contract",
-  [param("address").trim().isString(), query("tokenAddress").trim().isString(),
-  param("address").custom((value, { req }) => {
-    if (!web3.utils.isAddress(value)) {
-      throw new error("invalid address");
-    }
-    return true
-  }),
-],
-  async (req, res, next) => {
-   try {
-    const errors = validationResult(req);
-    if (!errors.isEmpty()) {
-     return res.status(400).json({ errors: errors.array() });
-    }
-    const { address = "*" } = req.params;
-    let { tokenAddress = "" } = { ...req.query };
-    const tokenContractAddress = prefixContractAddress(
-     tokenAddress.toString("hex")
-    );
-    if (!web3.utils.isAddress(tokenContractAddress)) {
-     return res.status(500).json({ error: "invalid contract address" });
+  router.get(
+    "/:address/symbol",
+    [param("address").trim().isString(), query("tokenAddress").trim().isString(),
+    param("address").custom((value, { req }) => {
+      if (!web3.utils.isAddress(value)) {
+        throw new error("invalid address");
+      }
+      return true
+    }),
+    ],
+    async (req, res, next) => {
+      try {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+          return res.status(400).json({ errors: errors.array() });
+        }
+        const { address = "*" } = req.params;
+        let { tokenAddress = "" } = { ...req.query };
+        const tokenContractAddress = prefixContractAddress(
+          tokenAddress.toString("hex")
+        );
+        if (!web3.utils.isAddress(tokenContractAddress)) {
+          return res.status(500).json({ error: "invalid contract address" });
+        }
```

Comment:
this may be handled in validators

Reply:
ok, will add all these kind of checks in validaton